vscode/src/edit/prompt/smart-apply/selection/custom-model.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    type CompletionParameters,
+    type EditModel,
+    type Message,
+    type ModelContextWindow,
+    PromptString,
+    TokenCounterUtils,
+    modelsService,
+    psDedent,
+} from '@sourcegraph/cody-shared'
+import * as vscode from 'vscode'
+import { getInstructionPromptWithCharLimit } from '../utils'
+import {
+    LLM_PARAMETERS,
+    SMART_APPLY_TOPICS,
+    type SelectionPromptProviderArgs,
+    type SelectionPromptProviderResult,
+    type SmartApplySelectionProvider,
+} from './base'
+
+export const SMART_APPLY_INSTRUCTION_TOKEN_LIMIT = 500
+
+const DEFAULT_SELECTION_PROMPT = {
```

Comment:
```suggestion
const SELECTION_PROMPT = {
```

nit: I thought we were modifying the current selection prompt, rename this to avoid confusion as the other prompt is already in `default.ts`